src/speculators/models/eagle.py

```diff
@@ -532,7 +529,45 @@ def _create_transformer_layer(self) -> nn.Module:
         return layer
 
     def _layernorm_class(self) -> type[nn.Module]:
-        return LlamaRMSNorm
+        return self._architecture_classes[1]
 
     def _transformer_layer_class(self) -> type[nn.Module]:
-        return LlamaDecoderLayer
+        return self._architecture_classes[0]
+
+    def _import_model_classes(self) -> tuple[type, type]:
+        """
+        Get the decoder layer class and layer normalization class for a given config
+        and decoder layer name. Uses the config qualified name to find the corresponding
+        modeling module to import the decoder layer and normalization classes from.
+        """
+        # e.g. LlamaConfig + LlamaDecoderLayer
+        # -> get qualname: transformers.models.llama.configuration_llama.LlamaConfig
+        # -> expected path: transformers.models.llama.modeling_llama.LlamaDecoderLayer
+        # -> expected path: transformers.models.llama.modeling_llama.LlamaRMSNorm
+
+        module_path = self.config.transformer_layer_config.__class__.__module__
+        parts = module_path.split(".")
+        if module_path.startswith("transformers.models.") and len(parts) >= 3:  # noqa: PLR2004
+            model_name = parts[2]
+        else:
+            raise ValueError(
+                "Could not import model classes. Please ensure the config is a valid "
+                "model specific config. i.e. LlamaConfig"
+            )
+
+        # Import the modeling module
+        modeling_module = importlib.import_module(
+            f"transformers.models.{model_name}.modeling_{model_name}"
+        )
+
+        transformer_arch_name = self.config.transformer_layer_architecture
```

Comment:
A nice to have here would be supporting an "auto" or "from_layer_config" reserved name that would match a ".*DecoderLayer" regex against the classes available in the module. Once found, it would update the conig with the proper value and use that class instance. The use case for this is for easier creation of the speculator model so only the transformer config needs to be determined and passed in ahead of time and we remove potential conflicting input args in the config.

Not critical, though, so if it's too much to include here, we can open up an issue for it to put it on the backlog